generate_diagrams.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+directory="./docs/model_diagrams/"
+
+if [ ! -d "$directory" ]; then
+    # If the directory doesn't exist, create it
+    mkdir -p $directory
+fi
```

Comment:
`-p` won't error if the directory already exists.
```suggestion
mkdir -p $directory
```

Reply:
f83c3632ae3a967745b22f528552865eb7be904b